### PR TITLE
Clean-up request IP validation

### LINKED ISSUE #21351: Webhook IP Denial Error after Upgrading to v10.9.0
### Describe the Bug

After upgrading our Directus instance from v10.8.3 to v10.9.0, our webhooks began failing with the error: "Requested URL resolves to a denied IP address". This issue was not present before the upgrade, and the webhook URLs have not changed. The problem causes all outbound webhook requests to fail, which is critical to our operations. Downgrading back to v10.8.3 resolves the issue, suggesting a regression or change in IP handling in the newer version.

Screenshot from 2024-02-07 08-47-14

#### Additional Context:
The issue was temporarily resolved by _downgrading_ to v10.8.3. We suspect there might be a change in how outbound IP addresses are handled or validated in the new version, which could be inadvertently blocking legitimate requests.

### To Reproduce

1. Upgrade Directus instance from v10.8.3 to v10.9.0.
2. Set up a webhook pointing to an external URL.
3. Trigger the webhook.
4. Observe that the webhook fails with the error message: "Requested URL resolves to a denied IP address".

### Directus Version

v10.9.0

### Hosting Strategy

Self-Hosted (Docker Image)

### PR DESCRIPTION
**Pull request by paescuj**
## Scope

What's changed:

- Cleaned-up the IP validation used for external requests
  - The remote address in response is usually `undefined` as the socket has already been terminated at that point [^1]
  - With the updated IP check in #20971, requests could have been blocked now because the check (reasonably enough) expects the IP to be defined, see #21351
  - By moving the validation from interceptors to a customized HTTP agent, we ensure the IP is available

## Potential Risks / Drawbacks

--

## Review Notes / Questions

IP validation can be tested with default or custom `IMPORT_IP_DENY_LIST` config [^2] via import, request operation in Flow or Webhook.

---

Fixes #21351

[^1]: https://nodejs.org/api/net.html#socketremoteaddress
[^2]: https://docs.directus.io/self-hosted/config-options.html#security

### CODE REVIEW

File: .changeset/chatty-jobs-scream.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@directus/api": patch
+---
+
+Cleaned-up IP validation on imports from URLs, requests in Flows and Webhooks to ensure the requests aren't incorrectly blocked
```
Comment (paescuj) [inline .changeset/chatty-jobs-scream.md:5]:
```suggestion
Cleaned-up and fixed IP validation used for external requests
```

**Review by br41nslug — COMMENTED** [review]
Working very well in my tests/experiments:
- DNS rebinding blocked ✅ 
- redirects to blocked ip's caught ✅ 

Just noticed 3 things we may want to address:
- `IMPORT_IP_DENY_LIST="127.0.0.1"` does not seem to block `localhost` but `"0.0.0.0"` does
- The errors in the console are huge (stacktrace)

https://github.com/directus/directus/assets/9389634/f829aa84-a582-4bbd-bfec-acfddc301147

- The API returns `503 Service Unavailable` where i would expect a 400 forbidden error

**Comment by paescuj** [comment]
>     * `IMPORT_IP_DENY_LIST="127.0.0.1"` does not seem to block `localhost` but `"0.0.0.0"` does

The thing is that localhost is usually getting resolved to `::1` and not to `127.0.0.1` which means technically it works as expected. Not sure if we want to treat this specially, as one can use `0.0.0.0` or `127.0.0.1,::1` for this case.

**Comment by br41nslug** [comment]
> The thing is that localhost is usually getting resolved to ::1 and not to 127.0.0.1

Fascinating this sounds like a bug to me, I explicitly do not have an IPv6 interface on my dev machine so that it resolves to `::1` makes little sense. (It does and including `::1` does block localhost it is however very unintuitive on a machine without IPv6 enabled)

**Comment by paescuj** [comment]
> The errors in the console are huge (stacktrace)

Thanks, fixed now in 73e1235 👍 

> > The thing is that localhost is usually getting resolved to ::1 and not to 127.0.0.1
> 
> Fascinating this sounds like a bug to me, I explicitly do not have an IPv6 interfaces on my dev machine so that it resolves to `::1` makes little sense.
> 
> In any case adding the `::1` does resolve the issue and indeed blocks `localhost`. We'll need to document this somewhere i think.

https://twitter.com/matteocollina/status/1539908299667161088 😄 

I'll add it to the documentation 👌

**Review by br41nslug — APPROVED** [review]

**Pull request merged**